arches/app/utils/transaction.py

```diff
@@ -0,0 +1,20 @@
+from arches.app.models.models import EditLog, TileModel, ResourceInstance
+from django.db import DatabaseError
+
+
+# Given a transaction ID, reverse (delete or update) tiles and resources created/updated during the transaction
+def reverse_edit_log_entries(transaction_id):
+    transaction_changes = EditLog.objects.filter(transactionid=transaction_id).order_by("-timestamp").all()
+    number_of_db_changes = 0
+    try:
+        for edit_log in transaction_changes:
+            if edit_log.edittype == "create":
+                number_of_db_changes += ResourceInstance.objects.filter(resourceinstanceid=edit_log.resourceinstanceid).delete()[0]
```

Comment:
we should use the proxy models for deleting and saving [resources](https://github.com/archesproject/arches/blob/22fbc5bd534b99a874bc9835e501cdf572742524/arches/app/models/resource.py#L58) and [tiles](https://github.com/archesproject/arches/blob/22fbc5bd534b99a874bc9835e501cdf572742524/arches/app/models/tile.py#L47) here, as that will also manage ES indexes, etc.